test/controllers/species_lists_controller_test.rb

```diff
@@ -91,18 +91,21 @@ def obs_params(obs, vote)
 
   ##############################################################################
 
+  def test_index
+    # get species_lists_path
+    # Above causes ActionController::UrlGenerationError:
+    # No route matches {:action=>"/species_lists", :controller=>"species_lists"}
```

Comment:
Why doesn't this work?

Reply:
And similar with `get species_lists_path` for :show.

Reply:
Hi Joe, 
from what you've got here, and from my own rake routes, `get species_list_path` seems to be working OK... remember the singular one needs an `:id`, so it should be `get species_list_path(:id)` and return `"/species_lists/:id"`

Reply:
No luck. The routes seem OK, but I still get an error. 
Could it be branch dependent, or depend on where you run `get species_lists_path`?

Here's what I see in the bootstrap4 base branch:
```ruby
vagrant@vagrant /vagrant/mushroom-observer (bootstrap-4)
$ rails t test/controllers/species_lists_controller_test.rb -n test_index
Started with run options -n test_index --seed 55618

ERROR["test_index", #<Minitest::Reporters::Suite:0x000055e5c403bd30 @name="SpeciesListsControllerTest">, 1.8551793299993733]
 test_index#SpeciesListsControllerTest (1.86s)
Minitest::UnexpectedError:         ActionController::UrlGenerationError: No route matches {:action=>"/species_lists", :controller=>"species_lists"}
            test/functional_test_case.rb:26:in `get'
            test/controllers/species_lists_controller_test.rb:102:in `test_index'
```
SpeciesListsControllerTest#test_index:
```ruby
  def test_index
    get species_lists_path
    assert_template(:index)
  end
```

Reply:
Yes! This is what has been driving me bonkers. 

It is what drove me on a 2-day detour to figure out how Rails recommends running controller tests, and that's how i landed on using a different class inheritance for IntegrationControllerTest instead of FunctionalTestCase.

However, I've been totally flummoxed trying to get that to work too. It does generate paths correctly, but then generates other errors, such as where it says :index and :show paths need a login. They don't.

Generally speaking, i'd have more faith investing time in the integration test approach, since this has gotten all the attention from the rails development team for the last 5 years. But it remains to be discovered, how to make it work in our app. I've only gotten a few tests working with it. And all our working tests are with the old way, so that carries more weight.

Still, maybe try inheriting from the new class for species_lists_controller and see if you can find the way?